src/App.tsx

```diff
@@ -1,26 +1,79 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from './hooks';
+import { setTodos } from './features/todos';
+import { getTodos } from './api';
+import { setQuery, setStatus } from './features/filter';
+import { Status } from './types/Status';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodos = useAppSelector(state => state.currentTodo);
+  const { query, status } = useAppSelector(state => state.filter);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(response => {
+        dispatch(setTodos(response));
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
it's better to move the logic inside the useEffect hook to a separate function and move useEffect to the end of the code

Reply:
done